code/game/atoms_health.dm

```diff
@@ -0,0 +1,293 @@
+	var/health_current
+	/// Maximum health for simple health processing. Use `get_max_health()` or `set_max_health()` to reference/modify.
+	var/health_max
+
+	/**
+	 * LAZY List of damage type resistance or weakness multipliers, decimal form. Only applied to health reduction. Use `set_damage_resistance()`, `remove_damage_resistance()`, and `get_damage_resistance()` to reference/modify.
+	 *
+	 * Index should be one of the `DAMAGE_` flags.
+	 * Value should be a multiplier that is applied against damage. Values below 1 are a resistance, above 1 are a weakness.
+	 * Value of `0` is considered immunity.
+	 */
+	var/list/health_resistances
+
+	/// Minimum damage required to actually affect health in `can_damage_health()`.
+	var/health_min_damage = 0
+
+/**
+ * Whether or not the atom has health. Checks for a value in `health_max`.
+ * Can be overridden for additional checks.
+ * Returns `TRUE` or `FALSE`.
+ */
+/atom/proc/has_health()
+	SHOULD_CALL_PARENT(TRUE)
+	return !!health_max
+
+/**
+ * Retrieves the atom's current health, or `null` if not using health
+ */
+/atom/proc/get_current_health()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return health_current
+
+/**
+ * Retrieves the atom's maximum health, or `null` if not using health.
+ */
+/atom/proc/get_max_health()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	return health_max
+
+/**
+ * Retrieves the atom's current damage, or `null` if not using health.
+ * If `use_raw_values` is `TRUE`, uses the raw var values instead of the `get_*` proc results.
+ */
+/atom/proc/get_damage_value(use_raw_values)
+	if (!has_health())
+		return
+	if (use_raw_values)
+		return health_max - health_current
+	else
+		return get_max_health() - get_current_health()
+
+/**
+ * Retrieves the atom's current damage as a percentage where `100%` is `1.00`.
+ * If `use_raw_values` is `TRUE`, uses the raw var values instead of the `get_*` proc results.
+ */
+/atom/proc/get_damage_percentage(use_raw_values)
+	if (!has_health())
+		return
+	var/max_health = use_raw_values ? health_max : get_max_health()
+	return round(get_damage_value(use_raw_values) / max_health, 0.01)
+
+/**
+ * Checks if the atom's health can be restored.
+ * Should be called before `restore_health()` in most cases.
+ * Returns `null` if health is not in use.
+ * NOTE: Does not include a check for death state by default, to allow repairing/healing atoms back to life.
+ */
+/atom/proc/can_restore_health(damage, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	if (!damage)
+		return FALSE
+	if (get_current_health() == get_max_health())
+		return FALSE
+	return TRUE
+
+/**
+ * Checks if the atom's health can be damaged.
+ * Should be called before `damage_health()` in most cases.
+ * Returns `null` if health is not in use.
+ */
+/atom/proc/can_damage_health(damage, damage_type = null)
```

Comment:
Unless the diff is incomplete this proc is not called anywhere.

Reply:
A case of something that was in use and then ended up not being used by the updated blob or rune wall code, but is something that will be used elsewhere when I incorporate this into other parts of the code.